plugin/evm/network_handler.go

```diff
@@ -21,38 +22,52 @@ import (
 var _ message.RequestHandler = &networkHandler{}
 
 type networkHandler struct {
-	stateTrieLeafsRequestHandler  *syncHandlers.LeafsRequestHandler
-	atomicTrieLeafsRequestHandler *syncHandlers.LeafsRequestHandler
-	blockRequestHandler           *syncHandlers.BlockRequestHandler
-	codeRequestHandler            *syncHandlers.CodeRequestHandler
-	signatureRequestHandler       *warpHandlers.SignatureRequestHandler
+	leafRequestHandlers     map[message.NodeType]*syncHandlers.LeafsRequestHandler
+	blockRequestHandler     *syncHandlers.BlockRequestHandler
+	codeRequestHandler      *syncHandlers.CodeRequestHandler
+	signatureRequestHandler *warpHandlers.SignatureRequestHandler
+}
+
+type LeafRequestTypeConfig struct {
+	NodeType     message.NodeType
+	NodeKeyLen   int
+	TrieDB       *triedb.Database
+	UseSnapshots bool
+	MetricName   string
 }
 
 // newNetworkHandler constructs the handler for serving network requests.
 func newNetworkHandler(
 	provider syncHandlers.SyncDataProvider,
 	diskDB ethdb.KeyValueReader,
-	evmTrieDB *triedb.Database,
-	atomicTrieDB *triedb.Database,
 	warpBackend warp.Backend,
 	networkCodec codec.Manager,
+	leafRequesTypeConfigs map[message.NodeType]LeafRequestTypeConfig,
```

Comment:
should we just take `map[message.NodeType]*syncHandlers.LeafsRequestHandler` here possibly it can be shortened using a type declaration like `type LeafHandlers map[message.NodeType]*syncHandlers.LeafsRequestHandler`